flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerGateway.java

```diff
@@ -112,22 +111,29 @@ void notifySlotAvailable(
 	 * Unregisters an infoMessage listener
 	 *
 	 * @param infoMessageListenerAddress address of infoMessage listener to unregister from this resource manager
-	 *
 	 */
 	void unRegisterInfoMessageListener(String infoMessageListenerAddress);
 
 	/**
 	 * shutdown cluster
+	 *
 	 * @param finalStatus
 	 * @param optionalDiagnostics
 	 */
 	void shutDownCluster(final ApplicationStatus finalStatus, final String optionalDiagnostics);
 
 	/**
 	 * Gets the currently registered number of TaskManagers.
-	 * 
+	 *
 	 * @param leaderSessionId The leader session ID with which to address the ResourceManager.
 	 * @return The future to the number of registered TaskManagers.
 	 */
 	Future<Integer> getNumberOfRegisteredTaskManagers(UUID leaderSessionId);
+
+	/*
+	 * Sends the heartbeat to resource manager from task manager
+	 *
+	 * @param resourceID unique id of the task manager
+	 */
+	void sendHeartbeatFromTaskManager(final ResourceID resourceID);
```

Comment:
clearer to name this method `heartbeatFromTaskManager` or `reportHeartbeatFromTaskManager`.